frontend/ui/uimanager.lua

```diff
@@ -218,6 +219,8 @@ function UIManager:close(widget, refreshtype, refreshregion)
     local dirty = false
     -- first send close event to widget
     widget:handleEvent(Event:new("CloseWidget"))
+    -- Ensure all the widgets can get onFlushSettings event.
+    widget:handleEvent(Event:new("FlushSettings"))
```

Comment:
Good catch. I was under the impression that it's doing broadcast. `handleEvent` is probably not too much overhead.

Would it be better to do flush settings first before the close event?